CourseGrab/ViewControllers/MainNavigationController.swift

```diff
@@ -30,10 +30,51 @@ class InteractivePopRecognizer: NSObject, UIGestureRecognizerDelegate {
     
 }
 
+// MARK: - BigHitNavigationBar
+
+private class BigHitNavigationBar: UINavigationBar {
+
+    private let tapOffset: CGFloat = 40
+
+    weak var navigationController: UINavigationController?
+
+    override func hitTest(_ point: CGPoint, with event: UIEvent?) -> UIView? {
+        if bounds.insetBy(dx: 0, dy: -tapOffset).contains(point), let item = navigationController?.topViewController?.navigationItem {
+            var candidateViews = [
+                item.backBarButtonItem?.customView,
+                item.leftBarButtonItem?.customView,
+                item.rightBarButtonItem?.customView
+                ].compactMap { $0 }
+            item.leftBarButtonItems?.compactMap { $0.customView }.forEach { candidateViews.append($0) }
+            item.rightBarButtonItems?.compactMap { $0.customView }.forEach { candidateViews.append($0) }
+
+            for view in candidateViews {
```

Comment:
can we use `forEach` here? just to be consistent w/ above

Reply:
That does not work because you cannot return a non-void value in a forEach closure.

Reply:
oh mb i didn't see that you were returning a value